core/options.js

```diff
@@ -250,11 +259,23 @@ Blockly.Options.prototype.getMetrics;
 Blockly.Options.parseMoveOptions_ = function(options, hasCategories) {
   var move = options['move'] || {};
   var moveOptions = {};
-  if (move['scrollbars'] === undefined && options['scrollbars'] === undefined) {
+  if (move['scrollbars'] === undefined) {
     moveOptions.scrollbars = hasCategories;
+  } else if (move['scrollbars'] &&
+      move['scrollbars']['horizontal'] !== undefined &&
+      move['scrollbars']['vertical'] !== undefined) {
+    moveOptions.scrollbars.horizontal = !!move['scrollbars']['horizontal'];
+    moveOptions.scrollbars.vertical = !!move['scrollbars']['vertical'];
```

Comment:
There's an opportunity here to normalize the internal options from the external options. 
Yes, we accept options in a number of various ways (scrollbar can be boolean or an object), but internally since we're setting the values here, is there a reason we couldn't normalize everything to use the new scrollbaroptions format?

Reply:
I started doing that at first, but I was worried about how moveOptions is public. So, I wanted to make it so that the type didn't change if the new single-scrollbar feature wasn't being used.

Reply:
Ok, two things: 
- This will only accept an object with both horizontal and vertical defined, can we default to false? if only one is defined.
- Let's see if we can simplify it a little, how about:

```js
if (move['scrollbars'] === undefined && options['scrollbars'] === undefined) {
   moveOptions.scrollbars = hasCategories;  
} else {
   if (typeof moveOptions['scrollbars'] === 'object') {
      moveOptions.scrollbars. horizontal = !!moveOptions['scrollbars']['horizontal'];
      moveOptions.scrollbars. vertical = !!moveOptions['scrollbars']['vertical'];
   } else {
       moveOptions.scrollbars = !!move['scrollbars'] || !!options['scrollbars'];
   }
}
```

Reply:
I've simplified a bit, but I've left in the logic that changes `scrollbars` to a `boolean` if `horizontal` and `vertical` are the same value. Leaving that logic in allows me to simplify checks for for whether scrollbars exist because I can do `!!moveOptions.scrollbars` (otherwise if both `horizontal` and `vertical` are set `false` in the object, this would return `true` incorrectly).

Reply:
Sounds good, please add a comment for why this is needed.

Reply:
added